rfc/2023/2023-05-01-va-rfc-frontend-performance.md

```diff
@@ -0,0 +1,172 @@
+
+### Have VFS teams set file size limits tailored to their applications
+
+["The Performance Inequality Gap," 2023](https://infrequently.org/2022/12/performance-baseline-2023/) suggests file size limits for HTML, CSS, and JavaScript, for example "~151KiB of HTML/CSS/fonts and ~300-350KiB of JavaScript (gzipped)". Using that report as a guide, we can determine reasonable file size limits and include them in our own performance budgets. VFS teams would use these baselines, along with the information about their application's real world performance provided by Datadog RUM and the Performance Report dashboard, to manage performance budgets for their applications.
+
+### Leverage existing tools to reduce code duplcation
+
+As mentioned above, we can leverage our existing build tool, Webpack, to [prevent duplication](https://webpack.js.org/guides/code-splitting/#prevent-duplication) of code, removing the multiple copies of the same libraries in bundles. Webpack also provides options for [notifying teams of performance issues](https://webpack.js.org/configuration/performance/), another way to help teams understand performance and adopt a budget.
+
+## Risks
+
+
+>List the risks of this approach
+> * There are always risks. What are the risks of this solution?
+> * These are the things people will bring up in opposition to your idea or plans. Acknowledge them.
+
+- There are large assets on the page such as Google Analytics and Digital Voice scripts that we cannot optimize or remove. Performance budgets can and should acknowledge this.
+- Removing duplicate scripts can create dependencies between bundles. Given our Webpack configuration already injects "shared modules", this dependency is already expected, we're just missing the de-duplication step.
+- Applications may require lots of refactoring to reach a performance budget goal, or teams may simply not have a lot of time to work on performance. Budgets can be adopted gradually, and process created for customizing budgets and allowing exceptions when there are significant barriers to increasing performance.
+- Teams may need help improving performance. We can adopt a "we're all in this together" approach and empower teams to help each other
+- This could be perceived as a Platform responsibility. If VFS teams are empowered to make or propose improvements, this work can and should be shared across VFS and Platform teams.
+- People may question the data informing this proposal. The VA-specific research mentioned is from 2019, and broadband adoption has likely changed since then. The guidance provided by "The Performance Inequality Gap" is based on global measures, not US or Veteran-specific measures. We could spend time gathering more data on the performance experiences of VA.gov users.
+
+## Alternatives
```

Comment:
I also reference some [potential performance improvements](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21595) previously. Some other options not listed here:

- Upgrading to using HTTP/2 (more connections allowed)
- Using better compression with Brotli

Reply:
Also added under `Alternatives`, though I could almost use a `Complementary` section for this and things like it.


Thanks for pointing me to that! Have you considered introducing your RFC to `va.gov-platform-architecture`